share/ruby-build/1.8.6-p383

```diff
@@ -1,3 +1,3 @@
 require_gcc
-install_package "ruby-1.8.6-p383" "http://cache.ruby-lang.org/pub/ruby/1.8/ruby-1.8.6-p383.tar.gz#cea6c67f737727007ec89d1c93fd7d0dba035220f981706091b8642d7a43c03a" auto_tcltk standard
-install_package "rubygems-1.3.7" "http://production.cf.rubygems.org/rubygems/rubygems-1.3.7.tgz#388b90ae6273f655507b10c8ba6bee9ea72e7d49c3e610025531cb8c3ba67c9d" ruby
+install_package "ruby-1.8.6-p383" "https://cache.ruby-lang.org/pub/ruby/1.8/ruby-1.8.6-p383.tar.gz#cea6c67f737727007ec89d1c93fd7d0dba035220f981706091b8642d7a43c03a" auto_tcltk standard
+install_package "rubygems-1.3.7" "https://rubygems.global.ssl.fastly.net/rubygems/rubygems-1.3.7.tgz#388b90ae6273f655507b10c8ba6bee9ea72e7d49c3e610025531cb8c3ba67c9d" ruby
```

Comment:
Like @sferik, I'm reluctant to switch official download locations away from `rubygems.org` domain and onto `fastly.net`. Even if the Rubygems team switches to Fastly going forward, are they going to at least provide canonical download resources under `rubygems.org` that result in a redirect to a CDN like Fastly?

In any case, this switch should be done as a separate commit because it's a separate topic that we might or might not accept at this point.